model/task/db.go

```diff
@@ -1232,9 +1216,7 @@ func FindUniqueBuildVariantNamesByTask(projectId string, taskName string, repoOr
 			{RevisionOrderNumberKey: bson.M{"$lte": repoOrderNumber}},
 		},
 	}
-	if !legacyLookup {
-		query[BuildVariantDisplayNameKey] = bson.M{"$exists": true, "$ne": ""}
-	}
+	query[BuildVariantDisplayNameKey] = bson.M{"$exists": true, "$ne": ""}
```

Comment:
nit: seems like you could absorb this into the `query` initialization right above this since it's not conditional anymore.

Reply:
Its no longer necessary so I removed it